cc/google/iot.py

```diff
@@ -176,35 +189,81 @@ def send_recipe_to_device_via_IoT(device_id, commands_list):
 
     # send a config message to a device
     config = {}  # a python dict
-    config['lastConfigVersion'] = str(version)
-    config['messageId'] = str(int(time.time()))  # epoch seconds as message ID
-    config['deviceId'] = str(device_id)
-    config['commands'] = commands_list
+    config["lastConfigVersion"] = str(version)
+    config["messageId"] = str(int(time.time()))  # epoch seconds as message ID
+    config["deviceId"] = str(device_id)
+    config["commands"] = commands_list
 
     config_json = json.dumps(config)  # dict to JSON string
-    print('send_recipe_to_device_via_IoT: Sending commands to device: {}' \
-          .format(config_json))
+    print(
+        "send_recipe_to_device_via_IoT: Sending commands to device: {}".format(
+            config_json
+        )
+    )
 
     config_body = {
-        'versionToUpdate': version,
-        'binaryData': base64.urlsafe_b64encode(
-            config_json.encode('utf-8')).decode('ascii')
+        "versionToUpdate": version,
+        "binaryData": base64.urlsafe_b64encode(config_json.encode("utf-8")).decode(
+            "ascii"
+        ),
     }
-    res = iot_client.projects().locations().registries().devices(
-        ).modifyCloudToDeviceConfig(
-            name=device_path, body=config_body).execute()
+    res = iot_client.projects().locations().registries().devices().modifyCloudToDeviceConfig(
+        name=device_path, body=config_body
+    ).execute()
     # print('config update result: {}'.format( res ))
 
-# ------------------------------------------------------------------------------
+
+def send_start_recipe_command(device_uuid, recipe_uuid, recipe_dict=None):
```

Comment:
Does this also handle sending a URL to the recipe.json that has been uploaded to a public bucket?